06.wc/wc.rb

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+CHARACTER_WIDTH = 8
+
+def main
+  option = ARGV.getopts('l')
+  file_tables = ARGV.size >= 1 ? read_argv : read_stdin
+  count_tables = count_file_tables(file_tables)
+  output_count_tables(option, file_tables, count_tables)
+end
+
+def read_argv
+  ARGV.map do |file|
+    file_table = {}
+    file_table[:lines] = File.readlines(file)
+    file_table[:file_name] = file
+    file_table
+  end
+end
+
+def read_stdin
+  file_table = {}
+  file_table[:lines] = $stdin.readlines
+  file_table[:file_name] = nil
+  [file_table]
+end
+
+def count_file_tables(file_tables)
+  file_tables.map do |file_table|
+    count_table = { line_count: 0, word_count: 0, byte_count: 0, file_name: file_table[:file_name] }
+    file_table[:lines].each do |line|
+      count_table[:line_count] += line.count("\n")
+      count_table[:word_count] += line.split.size
+      count_table[:byte_count] += line.bytesize
+    end
+    count_table
+  end
+end
+
+def output_count_tables(option, file_tables, count_tables)
+  count_tables.each do |count_table|
+    show_table(count_table, option)
+  end
+  return unless file_tables.size > 1
+
+  total_table = sum_count_tables(count_tables)
+  show_table(total_table, option)
+end
+
+def show_table(output_table, option)
+  if option['l']
+    puts "#{format_number(output_table[:line_count])} #{output_table[:file_name]}"
+  else
+    puts "#{format_number(output_table[:line_count])}"\
+          "#{format_number(output_table[:word_count])}"\
+          "#{format_number(output_table[:byte_count])} #{output_table[:file_name]}"
```

Comment:
無理に連結しようとせずに、putsを毎回読んだ方が読みやすいと思います〜。

```suggestion
    puts "#{format_number(output_table[:line_count])}"
    puts "#{format_number(output_table[:word_count])}"
    puts "#{format_number(output_table[:byte_count])} #{output_table[:file_name]}"
```

Reply:
putsを3回呼ぶ記述で試してみたのですが改行が入力されてしまい狙った表示になりませんでした。

<img width="562" alt="puts" src="https://user-images.githubusercontent.com/48541378/162642257-69933c31-da4e-4d4c-8292-8f457e8f67b4.png">

下記のように修正したところ正しく表示されたのですが可読性という面ではこれで良さそうでしょうか？

```
 print format_number(output_table[:line_count])
 print format_number(output_table[:word_count])
 print format_number(output_table[:byte_count])
 puts " #{output_table[:file_name]}"
```

以下の理由により自分では正しい記述なのか判断できませんでした。

* 連結は無くなったが1行増えた
* `print` と`puts` が混在している

とても細かい質問で申し訳ないのですが、Rubyの最後のプラクティスということもありどうしても気になるのでアドバイスお願いいたします！

Reply:
あ、そうそう、ここはputsじゃなくてprintが正解ですね。間違えました。すいません😓

> 以下の理由により自分では正しい記述なのか判断できませんでした。

この点は特に問題ないと思います！

Reply:
アドバイスありがとうございます！
スッキリしました！
修正してマージしておきます🙇‍♂️